### ISSUE TITLE
Caching across pods for Altinn Apps

### ISSUE THREAD

**Issue by acn-sbuad** [Altinn/app-lib-dotnet#170]
## Description
Currently, we do not have support for caching data across pods for an app. 
If autoscaling is introduces or replica count is inreaced we can have multiple instances of an app running in the cluster. 
how can we support caching across multiple instances of an app? 

## In scope
> What's in scope of this analysis?

Lorem ipsum

## Out of scope
> What's **out** of scope for this analysis?

Lorem ipsum

## Constraints
> Constraints or requirements (technical or functional) that affects this analysis.

Lorem ipsum

## Analysis

Lorem ipsum

## Conclusion
> Short summary of the proposed solution.

## Tasks
- [ ] Is this issue labeled with a correct area label?
- [ ] QA has been done

**Comment by altinnadmin** [comment]
I hope we can avoid that, given that the SPA is loaded once (and can cache data in the browser), and that we introduce capability for cache in Platform ref. Altinn/altinn-studio#5621.

Caching makes things more complex and should only be introduced if we must.

**Comment by elsand** [comment]
@altinnadmin [We are already using IMemoryCache](https://github.com/Altinn/altinn-studio/search?q=IMemoryCache), which is not distributed across pods. This can introduce subtle bugs due to cache incoherence. Also, the effect of this type of caching is questionable since afaik the LB doesn't pin sessions to any particular pod. Using IMemoryCache seems to me to be fundamentally incompatible with the architecture we're using, and should probably be removed alltogether if not replaced with some distributed cache like Redis.

**Comment by altinnadmin** [comment]
@elsand Yup, but only in Platform and we're looking into using redis there, ref. Altinn/altinn-studio#5621.

The only place that IMemoryCache seems to be used directly in apps [is for text resources](https://github.com/Altinn/altinn-studio/blob/9cb4f5008a6ec5854f905d9856c8c6f67631f624/src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.PlatformServices/Implementation/TextAppSI.cs#L70) (to avoid disk access), which is static across pods and does not change unless an app (pod) is redeployed.

**Comment by SandGrainOne** [comment]
Removing the platform label again. Platform have a separate issue for caching in platform.
